packages/scene-composer/src/components/ConvertSceneModal.tsx

```diff
@@ -0,0 +1,212 @@
+        });
+      }
+    } catch (e) {
+      setProgress(undefined);
+      setResult({ errorMessage: (e as Error).message });
+    }
+  }, [document, sceneLayerIds, sceneRootId, setSceneProperty, updateSceneNodeInternal]);
+
+  return (
+    <CenteredContainer
+      header={
+        <Header variant='h2'>
+          {formatMessage({ description: 'modal header text', defaultMessage: 'Convert scene' })}
+        </Header>
+      }
+      footer={
+        <Box float='right' padding={{ bottom: 's' }}>
+          {result ? (
+            <Button data-testid='ok-button' onClick={onClose}>
+              {formatMessage({ description: 'button label', defaultMessage: 'Ok' })}
+            </Button>
+          ) : (
+            <SpaceBetween size='s' direction='horizontal'>
+              <Button data-testid='cancel-button' onClick={onClose}>
+                {formatMessage({ description: 'button label', defaultMessage: 'Cancel' })}
+              </Button>
+
+              <Button data-testid='confirm-button' variant='primary' onClick={onConfirm} disabled={!!progress}>
+                {formatMessage({ description: 'button label', defaultMessage: 'Confirm' })}
+              </Button>
+            </SpaceBetween>
+          )}
+        </Box>
+      }
+    >
+      {!progress && !result && (
+        <Alert
+          type='warning'
+          header={formatMessage({ description: 'alert header text', defaultMessage: 'Converting scene' })}
+        >
+          {formatMessage({
+            description: 'alert body text',
+            defaultMessage:
+              'Proceeding with this action will permanently change your scene. You cannot undo this action',
+          })}
+        </Alert>
+      )}
+      {progress && <ConvertingProgress total={progress.total} converted={progress.converted} />}
+      {result &&
+        (isEmpty(result.failedNodes) && !result.errorMessage ? (
+          <Alert
+            type='success'
+            header={formatMessage({
+              description: 'alert header text',
+              defaultMessage: 'Convert to entities successful',
+            })}
+          >
+            {formatMessage({
+              description: 'alert body text',
+              defaultMessage: 'All nodes have been converted successfully',
+            })}
+          </Alert>
+        ) : (
+          <Alert
+            type='error'
+            header={formatMessage({ description: 'alert header text', defaultMessage: 'Convert to entities failed' })}
+          >
+            {result.errorMessage && <Box>{result.errorMessage}</Box>}
+
+            {!isEmpty(result.failedNodes) && (
+              <Box>
+                {formatMessage({ description: 'alert body text', defaultMessage: 'Some node conversion failed:' })}
+              </Box>
+            )}
+            {result.failedNodes &&
+              result.failedNodes.map((node) => (
+                <Box key={node.ref} padding={{ left: 's' }}>
+                  {node.name}
```

Comment:
What is the course of actions user can take towards failed nodes, can he/she retry failed node ?

Reply:
depending on the error. retry may help if it's a transient issue (like cannot update entity in UPDATING status), or need to update the node data (like property exceeds string length limit). this currently just shows the failed nodes. there will be a whole error handling story to handle those before GA